samcli/lib/telemetry/metric.py

```diff
@@ -158,6 +159,9 @@ def wrapped(*args, **kwargs):
             metric_specific_attributes["gitOrigin"] = get_git_remote_origin_url()
             metric_specific_attributes["projectName"] = get_project_name()
             metric_specific_attributes["initialCommit"] = get_initial_commit_hash()
+            # Event metrics
+            metric_specific_attributes["events"] = [e.to_json() for e in EventTracker.get_tracked_events()]
+            EventTracker.clear_trackers()
```

Comment:
would be good to make sure `clear_trackers` is called in the `track_command` test

Reply:
I'm not sure I understand. Since `clear_trackers` is called in `track_command`, any existing tests would call it by default, as all tests that call `track_command` will also call `clear_trackers`. Could you let me know which types of tests should be calling `clear_trackers`?

Reply:
My understanding is - `clear_trackers` must be called in `track_command`. 
If someone accidentally removes this line, we won't be able to catch it.

Reply:
With the additional `Event` unit tests I added in the latest commit (1b5efd5), if line 164 were to be accidentally deleted and things not function properly, one or more of the three test cases will fail, as previous Events will be carried over into the new tests (I tested on my machine with that line commented out, and that is the result I get). Another solution could be to leave a comment on that line, or update the `clear_trackers` docstring, as to provide more thorough reasoning for why the line is necessary where it is?

Reply:
> one or more of the three test cases will fail, as previous Events will be carried over into the new tests
We should avoid relying on this behavior. Each test should be standalone. States from previous test should not be carried over to next test.

Consider, if we create a test, and patch `EventTracker.clear_trackers` in that test. Then we'd be able to assert if `EventTracker.clear_trackers` is called. See [here](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called) for more info.

Reply:
I see, that would definitely help in ensuring the functionality remains the same. In addition, that would also remove the need for test ordering to be in a specific fashion. I will implement these changes as soon as I get the chance